python/ray/data/tests/test_consumption.py

```diff
@@ -1338,6 +1338,10 @@ def test_pandas_block_select():
 # tests should only be carefully reordered to retain this invariant!
 
 
+@pytest.mark.skip(
+    "The PyArrow version check caches results in a global variable. So, we don't raise "
```

Comment:
Is this flakiness new in this PR?

Reply:
Okay, so I looked more into it -- turns out to test is just really brittle. We check the PyArrow version once per process. Previously, we created readers in a remote task, so this wasn't an issue. 

I've rewritten the test to make it less brittle.